casepro/cases/models.py

```diff
@@ -258,37 +258,62 @@ def search(cls, org, user, search):
         return queryset.order_by('-opened_on')
 
     @classmethod
-    def get_or_open(cls, org, user, message, summary, assignee, user_assignee=None):
+    def get_or_open(cls, org, user, message, summary, assignee, user_assignee=None, contact=None):
+        """
+        Get an existing case, or open a new case if one doesn't exist. If message=None, then contact is required, and
+        any open case for that contact will be returned. If no open cases exist for the contact, a new case will be
+        created.
+        """
         from casepro.profiles.models import Notification
-
         r = get_redis_connection()
-        with r.lock(CASE_LOCK_KEY % (org.pk, message.contact.uuid)):
-            message.refresh_from_db()
 
-            # if message is already associated with a case, return that
-            if message.case:
-                message.case.is_new = False
-                return message.case
+        if message:
+            with r.lock(CASE_LOCK_KEY % (org.pk, message.contact.uuid)):
```

Comment:
Resolved in https://github.com/praekelt/casepro/pull/109/commits/e19477a834adff0e943e074f4532e02f4679fea0